engine/execution/ingestion/stop_at_height.go

```diff
@@ -0,0 +1,176 @@
+package ingestion
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine/execution/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type StopControl struct {
```

Comment:
since the logic has been extracted into this module, can we add more test cases?